lib/Runtime/Executor/ThreadPoolExecutor.h

```diff
@@ -25,6 +26,32 @@
 namespace glow {
 namespace runtime {
 
+/// This class implements a simple barrier with which to wait for all threads
+/// to exit a certain section of code before proceeding.
+class InflightBarrier final {
+public:
+  /// Decrement the count of threads in the barrier by \p decr.
+  void decrement(unsigned decr = 1);
+
+  /// Increment the count of threads in the barrier by \p incr.
+  void increment(unsigned incr = 1);
+
+  /// \returns the current count of the barrier.
+  unsigned count();
+
+  /// Wait for the barrier count to hit zero before continuing. This is
+  /// potentially a blocking call.
```

Comment:
It's definitely a blocking call, in fact all methods in this class are because you grab a lock.